nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/storage/PutGCSObject.java

```diff
@@ -322,199 +328,196 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
             return;
         }
 
-        final long startNanos = System.nanoTime();
+        try {
+            final long startNanos = System.nanoTime();
+
+            final String bucket = context.getProperty(BUCKET)
+                    .evaluateAttributeExpressions(flowFile)
+                    .getValue();
+            final String key = context.getProperty(KEY)
+                    .evaluateAttributeExpressions(flowFile)
+                    .getValue();
+            final boolean overwrite = context.getProperty(OVERWRITE).asBoolean();
+
+            final FlowFile ff = flowFile;
+            final String ffFilename = ff.getAttributes().get(CoreAttributes.FILENAME.key());
+            final Map<String, String> attributes = new HashMap<>();
+            final ResourceTransferSource resourceTransferSource = ResourceTransferSource.valueOf(context.getProperty(RESOURCE_TRANSFER_SOURCE).getValue());
```

Comment:
Please use the recently added `PropertyValue.asAllowableValue()` method:
```suggestion
            final ResourceTransferSource resourceTransferSource = context.getProperty(RESOURCE_TRANSFER_SOURCE).asAllowableValue(ResourceTransferSource.class);
```